sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_extensions.cs

```diff
@@ -0,0 +1,367 @@
+        {
+            if (hub == null)
+            {
+                throw new ArgumentNullException(nameof(hub));
+            }
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            if (endpoint == null)
+            {
+                throw new ArgumentNullException(nameof(endpoint));
+            }
+
+            this.credential = credential;
+            this.hub = hub;
+            this.endpoint = endpoint;
+
+            options ??= new WebPubSubServiceClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            apiVersion = options.Version;
+
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() },
+                new HttpPipelinePolicy[] { new WebPubSubAuthenticationPolicy(credential) },
+                new ResponseClassifier()
+            );
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebPubSubServiceClient"/>.
+        /// </summary>
+        /// <param name="connectionString">Connection string contains Endpoint and AccessKey.</param>
+        /// <param name="hub">Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.</param>
+        public WebPubSubServiceClient(string connectionString, string hub) : this(ParseConnectionString(connectionString), hub)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebPubSubServiceClient"/>.
+        /// </summary>
+        /// <param name="connectionString">Connection string contains Endpoint and AccessKey.</param>
+        /// <param name="hub">Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.</param>
+        /// <param name="options"></param>
+        public WebPubSubServiceClient(string connectionString, string hub, WebPubSubServiceClientOptions options) :
+            this(ParseConnectionString(connectionString), hub, options)
+        {
+        }
+
+        private WebPubSubServiceClient((Uri Endpoint, AzureKeyCredential Credential) parsedConnectionString, string hub) :
+            this(parsedConnectionString.Endpoint, hub, parsedConnectionString.Credential)
+        {
+        }
+
+        private WebPubSubServiceClient((Uri Endpoint, AzureKeyCredential Credential) parsedConnectionString, string hub, WebPubSubServiceClientOptions options) :
+            this(parsedConnectionString.Endpoint, hub, parsedConnectionString.Credential, options)
+        {
+        }
+
+        /// <summary>Broadcast message to all the connected client connections.</summary>
+        /// <param name="message"></param>
```

Comment:
Add a comment saying this is a JSON message?  Should we rename the parameter to `jsonMessage` to make that even more explicit?